### ISSUE TITLE
Misleading documentation for `IExportableTransaction.getReferenceNumber`

### ISSUE THREAD

**Issue by WorkProg** [ZUGFeRD/mustangproject#504]
https://github.com/ZUGFeRD/mustangproject/blob/6dcf7f50d9b64e3abce056e9db11cfb348896c8b/library/src/main/java/org/mustangproject/ZUGFeRD/IExportableTransaction.java#L324-L330

This is used for `BuyerReference` BT-10, and according to at least the German norm that seems like a strange documentation:

> eine vom Käufer zugewiesene und für internes Routing benutzte Kennung

> an identifier assigned by the buyer and used for internal routing

**Comment by jstaerk** [comment]
ok can you issue a PR?

**Comment by WorkProg** [comment]
sure #508

**Comment by Malove98** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.

**Comment by jstaerk** [comment]
Sorry only when looking th the PR I realized it's actually wrong. buyerReference is really more "your reference", mostly in terms of "care of", in XRechnung used for the recipient ID (leitweg id). When something refers to a previous order, like a cancellation or a corrected invoice it should rather use a InvoiceReferencedDocument, like correctedInvoices do in mustang.

**Comment by jstaerk** [comment]
no, getReferenceNumber()  refers to the buyer reference, you put the documentation for InvoiceReferencedDocument

**Comment by WorkProg** [comment]
Sorry, I'm just really confused what you are trying to tell me. This is the documentation I proposed:
```java
/**
 * get BuyerReference (BT-10) an identifier assigned by the buyer and used
 * for internal routing. Used for the Leitweg-ID.
 * 
 * @return the BuyerReference of this document
 */
```

Because currently, it is incorrectly documented to be used for Invoice Corrections: (red is the current, green my change):

![image](https://github.com/user-attachments/assets/45403551-0831-47ab-873e-474bf508cdfe)
